pre_commit/git.py

```diff
@@ -159,7 +166,8 @@ def git_path(name, repo='.'):
 
 def check_for_cygwin_mismatch():
     """See https://github.com/pre-commit/pre-commit/issues/354"""
-    if sys.platform in ('cygwin', 'win32'):  # pragma: no cover (windows)
+    if sys.platform in ('cygwin', 'win32') and \
```

Comment:
I believe this also needs to include `'msys'` for this check to be correct

Also I believe this check needs to get copied into the hook template since we start to do some `git` things in there as well

I don't think this check should include `is_mingw()` in it -- the warning is actually correct there.